### DISCUSSION TITLE
Web Server Creation for Novice AWS Administrators

### DISCUSSION THREAD

**Discussion by Squirrelbane75** [community/community#123155] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Show & Tell

### Body

Hello all!

I wanted to share a project I've been working on this semester. It involves quickly creating a LAMP Stack for a web server within AWS (Amazon Web Services). Programmers and technologists spend a lot time spinning up many web servers both for testing and production. I created this method so that web servers could be created much quicker, so that time can be spent doing more important things. 

For this particular stack, I am using Amazon Linux 2 as my Linux operating system distribution, Apache for the web server, MariaDB for the database, and php for the dynamic programming language. 

I've listed in the documentation how to go about the process, but I will give a quick overview. You'll first need to create an account on AWS and login. Next, search EC2 in the console and launch an EC2 instance. You will then name your instance and choose your Linux distribution (in this case Amazon Linux 2). Then, create a security key. You will need this key to ssh into your EC2 instance. The type of key you choose depends on the ssh method you are using (.pem for OpenSSH and .ppk for putty). Then, create a security group. This is a collection of firewalls that blocks or allows traffic from different sources to access your webpage. You will want to enable all traffic for testing your webpage. You can change the permissions within the security group at any time. Then, you will go find the "User Data" box under the "Advanced Details" tab. You can insert whatever code you want into the User Data box. This code will execute once within your instance upon bootup. I've attached the script that you will need to insert into the user data box. This script will automatically install and configure Apache, MariaDB, and php. It will also edit all of the permissions so that you can immediately begin working on your webpage without being the root user. 

Please leave any comments or reviews you wish to leave! The next step is to figure out how to automatically push the webservers out to other people as the admin. 

[How to setup an Amazon Linux 2 EC2 Server with automated LAM.docx](https://github.com/community/community/files/15265739/How.to.setup.an.Amazon.Linux.2.EC2.Server.with.automated.LAM.docx)
[LAMP Installation Script.txt](https://github.com/community/community/files/15265740/LAMP.Installation.Script.txt)

Hope this helps, thank you!

**Comment by iTechTop** [comment]
👋 Thank you for sharing your project! It sounds interesting and definitely useful for those frequently working with LAMP stack on AWS.

Your document describes the process nicely, and the script looks good too. Overall, it seems well thought out and concisely written. I think it could benefit from a couple of improvements to make it more user-friendly and inclusive.

1. Graphical Overview: Even though you provided excellent documentation, visual aids often help clarify complex procedures. Consider drawing diagrams or flowcharts showing the navigation and relationship between AWS Console menus, buttons, and inputs. This approach will greatly help non-native English speakers and beginners unfamiliar with AWS terminology.

2. Prerequisites Section: Begin with a brief introduction covering the assumed reader background, like knowing what AWS is, being comfortable navigating cloud consoles, and having experience working with Linux distributions. List any additional requirements, like having Puttygen for generating ppk keys.

3. Security Group Settings: Expand on the reasons for choosing "allow all traffic" initially. Emphasize that users should limit access to the minimum required for their scenario and recommend locking down unnecessary ports afterward.

4. Post-Setup Tips: Offer advice on best practices like keeping track of security groups, password rotation, patch management, and auditing. Discuss setting up Elastic IPs for fixed IP addresses, snapshotting volumes for backup, and monitoring charges incurred during experimentation.

5. Accessibility: Optimize your documents for PDF readers, screen readers, and mobile devices. Break large paragraphs into smaller chunks, use headings consistently, and emphasize essential points with bold or italic font styles.

Overall, congratulations on completing this project! Users will undoubtedly appreciate saving time and energy setting up web servers. Taking the extra steps to polish the documentation will broaden your audience and make your project shine even brighter. Best of luck figuring out the auto-deployment part, and feel free to ask questions if you face any hurdles!

**Reply by ebndev** [reply]
Hi @monolab825, thanks for wanting to participate in Discussions!

Please note: We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. 

Thank you for helping us maintain an authentic and beneficial space for everyone.
